pkg/eth/trace.go

```diff
@@ -1,67 +1,80 @@
 package eth
 
 import (
-	"bytes"
 	"cmp"
+	"context"
 	"encoding/json"
 	"fmt"
-	"io"
 	"math/big"
 	"net/http"
 
-	"github.com/duoxehyon/mev-share-go/types"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	ethrpc "github.com/ethereum/go-ethereum/rpc"
 )
 
 type TraceClient struct {
-	httpClient *http.Client
-	rpcURL     string
+	rpcClient *ethrpc.Client
 }
 
-func NewTraceClient(httpClient *http.Client, rpcURL string) *TraceClient {
-	return &TraceClient{
-		httpClient: httpClient,
-		rpcURL:     rpcURL,
+func NewTraceClient(ctx context.Context, httpClient *http.Client, rpcURL string) (*TraceClient, error) {
+	rpcClient, err := ethrpc.DialOptions(ctx, rpcURL, ethrpc.WithHTTPClient(httpClient))
+	if err != nil {
+		return nil, fmt.Errorf("dial options: %w", err)
 	}
+
+	return &TraceClient{
+		rpcClient: rpcClient,
+	}, nil
 }
 
-func (c *TraceClient) DebugTraceTransaction(txHash string) (CallFrame, error) {
-	payload := map[string]interface{}{
-		"method":  "debug_traceTransaction",
-		"id":      1,
-		"jsonrpc": "2.0",
-		"params": []interface{}{
-			txHash,
-			map[string]interface{}{
-				"tracer": "callTracer",
-				"tracerConfig": map[string]interface{}{
-					"withLog": true,
-				},
+func (c *TraceClient) DebugTraceTransaction(ctx context.Context, txHash string) (CallFrame, error) {
+	const (
+		method = "debug_traceTransaction"
+		tracer = "callTracer"
+	)
+	var callResp json.RawMessage
+	if err := c.rpcClient.CallContext(ctx, &callResp,
```

Comment:
Why don't use `CallFrame` here for parsing response instead of make another `json.Unmarshal` after request?

Reply:
updated in https://github.com/KyberNetwork/tradinglib/pull/65/commits/be362697cfebc26dcf947540e92dc89402614698